esphome/components/web_server/__init__.py

```diff
@@ -22,18 +24,36 @@
 web_server_ns = cg.esphome_ns.namespace("web_server")
 WebServer = web_server_ns.class_("WebServer", cg.Component, cg.Controller)
 
+
+def default_url(config):
+    if config[CONF_VERSION] == 1:
```

Comment:
```suggestion
    config = config.copy()
    if config[CONF_VERSION] == 1:
```

copy the input first, otherwise we end up changing the validator's state which can cause problems

Reply:
Changes done